tools/test-all

```diff
@@ -14,20 +14,51 @@ function run {
     fi
 }
 
-run ./tools/clean-repo
-run ./tools/test-tools
-run ./tools/lint-all --pep8
-run ./tools/test-migrations
-run ./tools/test-js-with-node
-run ./tools/run-mypy
-run ./tools/test-backend
-run ./tools/test-js-with-casper
-# Not running management test since it takes 40s and thus is too slow to be worth it.
-# run ./tools/test-management
-# Not running queue worker reload tests since it's low-churn code
-# run ./tools/test-queue-worker-reload
-# Not running documentation tests since it takes 20s and only tests documentation
-# run ./tools/test-documentation
-run ./tools/test-help-documentation.py
+function containsElement {
+    local e
+    for e in "${@:2}"; do [[ "$e" == "$1" ]] && return 0; done
+    return 1
+}
+
+TEST_COMMANDS=(
+    'run ./tools/check-provision' \
+    'run ./tools/clean-repo' \
+    'run ./tools/test-tools' \
+    'run ./tools/lint-all --pep8' \
+    'run ./tools/test-migrations' \
+    'run ./tools/test-js-with-node' \
+    'run ./tools/run-mypy' \
+    'run ./tools/test-backend' \
+    'run ./tools/test-js-with-casper' \
+    # Not running management test since it takes 40s and thus is too slow to be worth it.
+    # run ./tools/test-management
+    # Not running queue worker reload tests since it's low-churn code
+    # run ./tools/test-queue-worker-reload
+    # Not running documentation tests since it takes 20s and only tests documentation
+    # run ./tools/test-documentation
+    'run ./tools/test-help-documentation.py'
+)
+
+HAS_FORCE_ARGUMENT_TEST_COMMANDS=(
+    'run ./tools/lint-all --pep8' \
+    'run ./tools/test-backend' \
+    'run ./tools/test-js-with-casper'
+)
+
+if [[ $1 = "--force" ]]; then
```

Comment:
Also, I think this could be simplified - instead of going through two lists, you can `$eval` the first one, after removing tests that can have `--force` argument, and then go through the second one with the `if` statement. The only question is if the order of `lint-all` is crucial here and can we put it after `run-mypy` - have you discussed this approach?